tools/ConfigurationFilesGenerator/Program.cs

```diff
@@ -0,0 +1,385 @@
+            {
+                rules.Add(new AnalyzerRule(diagnostic.Id, diagnostic.Title.ToString(CultureInfo.InvariantCulture), diagnostic.HelpLinkUri, diagnostic.IsEnabledByDefault, diagnostic.DefaultSeverity, diagnostic.IsEnabledByDefault ? diagnostic.DefaultSeverity : null));
+            }
+        }
+    }
+
+    if (rules.Count > 0)
+    {
+        var sb = new StringBuilder();
+        sb.AppendLine("# global_level must be higher than the NET Analyzer files");
+        sb.AppendLine("is_global = true");
+        sb.AppendLine("global_level = 0");
+
+        var currentConfiguration = GetConfiguration(configurationFilePath);
+
+        if (currentConfiguration.Unknowns.Length > 0)
+        {
+            foreach (var unknown in currentConfiguration.Unknowns)
+            {
+                sb.AppendLine(unknown);
+            }
+        }
+        else
+        {
+            sb.AppendLine();
+        }
+
+        foreach (var rule in rules.OrderBy(rule => rule.Id).ThenBy(rule => rule.Title))
+        {
+            var currentRuleConfiguration = currentConfiguration.Rules.FirstOrDefault(r => r.Id == rule.Id);
+            var severity = currentRuleConfiguration != null ? currentRuleConfiguration.Severity : rule.DefaultEffectiveSeverity;
+
+            sb.AppendLine($"# {rule.Id}: {rule.Title}");
+            if (!string.IsNullOrEmpty(rule.Url))
+            {
+                sb.AppendLine($"# Help link: {rule.Url}");
+            }
+
+            sb.AppendLine($"# Enabled: {rule.Enabled}, Severity: {GetSeverity(rule.DefaultSeverity)}");
+
+            if (currentRuleConfiguration?.Comments.Length > 0)
+            {
+                foreach (var comment in currentRuleConfiguration.Comments)
+                {
+                    sb.AppendLine(comment);
+                }
+            }
+
+            sb.AppendLine($"dotnet_diagnostic.{rule.Id}.severity = {GetSeverity(severity)}");
+            sb.AppendLine();
+        }
+
+        if (File.Exists(configurationFilePath))
+        {
+            if ((await File.ReadAllTextAsync(configurationFilePath, cancellationToken)).ReplaceLineEndings() == sb.ToString().ReplaceLineEndings())
+            {
+                return;
+            }
+        }
+
+        configurationFilePath.CreateParentDirectory();
+        await File.WriteAllTextAsync(configurationFilePath, sb.ToString(), cancellationToken);
+        _ = Interlocked.Increment(ref writtenFiles);
+
+        static string GetSeverity(DiagnosticSeverity? severity)
+        {
+            return severity switch
+            {
+                null => "none",
+                DiagnosticSeverity.Hidden => "silent",
+                DiagnosticSeverity.Info => "suggestion",
+                DiagnosticSeverity.Warning => "warning",
+                DiagnosticSeverity.Error => "error",
+                _ => throw new Exception($"Severity '{severity}' is not supported"),
+            };
+        }
+    }
+});
+
+return writtenFiles;
```

Comment:
This is my last comment. When writtenFiles > 0, this is how it looks in the console output

![image](https://github.com/gsoft-inc/wl-dotnet-codingstandards/assets/14242083/766910b7-ad99-4c05-94aa-72b91db22236)

Maybe we could print something prior to returning the value, explaining that according to the currently installed analyzers and .NET SDK, there are X .editorconfig files waiting to be updated using the generator tool > which means that someone has to manually launch the tool then open a new PR